### DISCUSSION TITLE
Pivot table without id column

### DISCUSSION THREAD

**Discussion by justrustyspoons** [owen-it/laravel-auditing#777] [Q&A] [answered]
| Q                 | A
| ----------------- | ---
| Bug?              | yes
| New Feature?      | no
| Framework         | Laravel
| Framework version | 6.x.y
| Package version   | 9.x.y
| PHP version       | 7.3.y

#### Actual Behaviour

When updating a pivot table without an id column it throws an error that the auditable_id cannot be null

SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'auditable_id' cannot be null 

#### Expected Behaviour

Insert without auditable_id for pivot tables

#### Steps to Reproduce

calling $model->relationship()->updateExistingPivot($relationship_id, $data) in a controller throws the error SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'auditable_id' cannot be null 

#### Possible Solutions

Update migration to nullableMorphs('auditable') however this results in null values being recorded for old_values. Needs to record old_values through relationship instead of by pivot id

**Comment by abordage** [comment, marked as answer]
> Update migration to nullableMorphs('auditable') however this results in null values being recorded for old_values.

Instead, you can add auto-incrementing primary key and  `public $incrementing = true;` on pivot

**Reply by Virtute90** [reply]
> > Update migration to nullableMorphs('auditable') however this results in null values being recorded for old_values.
> 
> Instead, you can add auto-incrementing primary key and `public $incrementing = true;` on pivot

This solved my problem

**Comment by GytisLaukaitis** [comment]
And how about dettaching things ? :)

**Comment by anugrahdputra** [comment]
> > Update migration to nullableMorphs('auditable') however this results in null values being recorded for old_values.
> 
> Instead, you can add auto-incrementing primary key and `public $incrementing = true;` on pivot

Can I use laravel auditing with parameter $incrementing = false? because the id in the table uses a string type. Thanks
